curriculum/challenges/english/25-front-end-development/lab-football-team-cards/66e7ee20b79186306fc12da5.md

```diff
@@ -0,0 +1,763 @@
+---
+id: 66e7ee20b79186306fc12da5
+title: Build Football Team Cards
+challengeType: 14 
+dashedName: lab-football-team-cards
+# demoType can either be 'onClick' or 'onLoad'. If the project or lab doesn't have a preview, delete the property
+demoType: onClick
+---
+
+# --description--
+
+In this lab, you will write the functionality of adding data to the page from a JavaScript object, and of filtering dinamically these datas.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should create a `footballTeam` object, this object should have properties `team`, `year`, `headCoach`, `players`.
+1. The `team` property should contain the name of the team as a string.
+1. The `year` property should contain the year as a number.
+1. The `headCoach` property should contain the name of the head coach as a string.
+1. The `players` property should be an array containing at least four elements.
+1. Each element in the `players` array should be an object with properties `name`, `position`, `isCaptain`.
+1. The `name` property should contain the name of the player as a string.
+1. The `position` property should have one of the following values: `"forward"`, `"midfielder"`, `"defender"`, or `"goalkeeper"`.
+1. The `isCaptain` property should have value of a boolean, and it should be `true` only once between all the players.
+1. You should display the values from the object on the page.
+1. You should implent a filtering function: when the dropdown menu is used to select one of the positions, only players of that position should be shown.
+
+# --hints--
+
+You should have a `footballTeam` variable.
+
+```js
+assert.exists(footballTeam);
+```
+
+The `footballTeam` variable should be an object with four properties: `team`, `year`, `headCoach` and `players`.
+
+```js
+assert.isObject(footballTeam);
+assert.hasAllKeys(footballTeam, ['team', 'year', 'headCoach', 'players']);
+// it could be containsAllKeys if we want to give freedom to have more than these keys
```

Comment:
yeah, I think that should be fine to give them the freedom to have additional keys if they want. 
so, as long as they have the 4 keys we are looking for them it is fine if they want to add more